### DISCUSSION TITLE
allowBackup is not working on Android

### DISCUSSION THREAD

**Discussion by PouyaLitkouhi** [react-native-async-storage/async-storage#619] [General]
## Current behavior

Hello, on Android I want to keep user data after uninstalling to be able to use it on the next install. I'm using allowBackup and fullBackupContent. but each time I removed the app the data will be lost. I'm not sure what I'm missing

this is my AndroidManifest

`<application
      android:name=".MainApplication"
      android:label="@string/APPLICATION_NAME"
      android:icon="@mipmap/ic_launcher"
      android:allowBackup="true"
      android:theme="@style/AppTheme"
      android:fullBackupContent="@xml/my_backup_rules"
      tools:replace="android:fullBackupContent"
      android:hasFragileUserData="true">`

and this is my **my_backup_rules** file

`<?xml version="1.0" encoding="utf-8"?>
<full-backup-content>
   <include domain="database" path="RKStorage"/>
</full-backup-content>`

## Expected behavior

Data should persist after re-installing the app

## Repro steps

 1 - open the app and save something in asyncStorage
 2- uninstall the app
 3- reinstall it
 4- the previously saved data is lost

## Environment

- **Platforms tested**:
  - [*] Android
- **AsyncStorage version**: 1.15.4

**Comment by krizzu** [comment]
Hey,

From the look at your setup, it should properly do the backup. Are you triggering the backup, so that Android has a chance to save your database?

OS cannot save your files if app is in the foreground (unless you explicitly opt-in). I would suggest to do save some data to AsyncStorage, then [trigger backup process](https://developer.android.com/guide/topics/data/testingbackup#TestingBackup) to see if it's actually backed up.

**Reply by PouyaLitkouhi** [reply]
Hey,

How can I do that?
I saved some data in asyncStorage then after uninstalling and re-installing that data is missed.
I tried the command in the above link to see if the back up is working
`adb shell bmgr backup @pm@ && adb shell bmgr run`
`adb shell bmgr fullbackup <PACKAGE>`
It seems it's working.
<img width="757" alt="Screen Shot 2021-06-03 at 3 07 11 PM" src="https://user-images.githubusercontent.com/75396330/120718527-46f02a00-c47e-11eb-8ce3-a0c13e727c37.png">

I want to be able to retrieve some data that I stored in asyncStorage after the user re-installed the app. I'm not sure what I'm missing here, do I need to run something on app start or something like that?
I thought after the above setup it should work the same as when we use **android:hasFragileUserData** on Android 10+

**Reply by krizzu** [reply]
According to the docs, the restore should happen after you install the app and you launch it. [Have you tried testing restore?](https://developer.android.com/guide/topics/data/testingbackup#TestingRestore) This will simulate restoration of the backup, so if that works, I'd assume the app backup/restore would kick in.

**Reply by PouyaLitkouhi** [reply]
Thanks, I was finally able to get the setup works. **BUT**
1 - it's only working after I run 
`adb shell bmgr backup @pm@ && adb shell bmgr run`
`adb shell bmgr fullbackup <PACKAGE>`
then it will retrieve that back up each time I re-install the app. so still it won't get back-up automatically when the user uninstalls the app. is it possible to have the latest state of the app right before uninstalling the app?

2- Another issue that I had was with the package name, for some reason we need to use two different package names in our codebase. so the one that is in Androidmanifest is different from the one that I have when the app is installed and that makes the issue in the first place and I had to change it to use only one unified package name. do you have any idea to surpass this issue and still have two separate package names? 

Again thank you for the helps
